lsp-mode.el

```diff
@@ -8091,6 +8091,11 @@ such."
       (lsp--warn "No LSP server for %s(check *lsp-log*)." major-mode)
       nil)))
 
+(defun lsp--any-inconsistency? ()
+  "Check for any inconsistency on LSP package"
+  (and lsp-use-plists
+       (equal (symbol-function 'lsp-merge) 'ht-merge)))
```

Comment:
these are not the inconsistencies that we are going to hit. The issues will be when a package e. g. lsp-dart has version which has been compiled with lsp-plists off or if you update all at the same time but package.el decides first to download lsp-dart, compiles it against old lsp-mode and then updates lsp-mode. Unfortunately there is no easy way to fix that.

Reply:
Oh, I see the problem...
Do you see a way to check that or should we close this PR?

Reply:
one way to do that is to define in each package `(lsp-interface PACKAGE-check ...)` and then check if lsp-make-PACKAGE-check will return plist. IMO it is too much effort. It will be easier and less convenient for the users if we add a flag and ask users to delete all lsp-mode related packages and reinstall them and then set that flag to nil to skip that request.

Reply:
Yeah, I agree, but do you think the delete would be manual by the user or some that we should do if the flag is true?
And the set of the flag to true would be the user that would set it or we somehow?

Reply:
The flag will be something like: `lsp-skip-protocol-plist-warning` . IMO it is not a good idea to try to implement the delete automatically since there are different package managers with different APIs.

Reply:
Yeah, it was what I thought about it, but yet I think it's too bad for the UE, I see users complaining about having to set a variable manually because a package migration and users that never had to set anything manually needing to do that.
Maybe let the problem happen and have a well-documented page explaining about it and how to fix it could be better, WDYT?

Reply:
Maybe we can also explain this in the issue related to this PR and tag more maintainers to see their opinions

Reply:
> Maybe we can also explain this in the issue related to this PR and tag more maintainers to see their opinions

That won't work - we will get dozens of issue reports for months. We were in that situation with dash in the past. Dash had introduced new destructuring which was used in the project and that was reported in gitter/reddit/dap-mode repo/lsp-mode repo maybe 30 times. And lsp-mode was not that popular.

We may keep the setting in lsp-session so we can store it in lsp-session-file so once answered it won't be asked again.

Reply:
I see... Ok, I think saving in the lsp-session could be better, but yet we will get issues from users that ignored that, but I think that there is nothing that we can do more about it.
I'll implement this on `lsp-session` so

Reply:
@yyoncho done